lib/perl/Genome/Config/AnalysisProject/SubjectMapping/Command/Predict/SomaticValidation.pm

```diff
@@ -0,0 +1,163 @@
+            default => 'name',
+            valid_values => ['name','id'],
+        },
+    ],
+};
+
+sub help_brief {
+    return 'Predict the sample pairings needed to create Somatic Validation models for an AnalysisProject';
+}
+
+sub help_synopsis {
+    return "genome config analysis-project subject-mapping predict somatic-validation <analysis_project_id> <TSV file path>";
+}
+
+sub help_detail {
+    return <<"EOS"
+This command will predict subject mapping for Somatic Validation model types across all samples in an AnalysisProject.
+
+The output file is a 6 column, tab separated format with the following columns:
+
+tumor_subject normal_subject snv_variant_list_id indel_variant_list_id sv_variant_list_id tag
+
+EOS
+}
+
+sub valid_statuses {
+    return ("Pending", "Hold", "In Progress");
+}
+
+my @headers = ('tumor_sample', 'normal_sample','snv_variant_list_id', 'indel_variant_list_id', 'sv_variant_list_id','tag');
+
+sub execute {
+    my $self = shift;
+
+    my $analysis_project = $self->analysis_project;
+
+    my $normal_sample_common_names = $self->normal_sample_common_names;
+    my $tumor_sample_common_names  = $self->tumor_sample_common_names;
+
+    my $id_method = $self->sample_identifier;
+
+    my @all_dna_samples = grep {$_->sample_type =~ /dna/i} $analysis_project->samples;
+    my %dna_samples = map { $_->id => $_ } @all_dna_samples;
+    my @dna_sample_ids = sort keys %dna_samples;
+    $self->status_message('Found '. scalar(@dna_sample_ids) .' DNA samples');
+
+    my %dna_samples_by_individual_id;
+    for my $dna_sample_id (@dna_sample_ids) {
+        my $dna_sample = $dna_samples{$dna_sample_id};
+        push @{$dna_samples_by_individual_id{$dna_sample->individual->id}}, $dna_sample;
+    }
+    my @individual_ids = sort keys %dna_samples_by_individual_id;
+    $self->status_message('Found '. scalar(@individual_ids) .' indivudals');
+
+    my @subject_mappings;
+    for my $individual_id (@individual_ids) {
+        my $individual = Genome::Individual->get($individual_id);
+        my @individual_dna_samples = @{$dna_samples_by_individual_id{$individual_id}};
+        my $individual_dna_sample_count = scalar(@individual_dna_samples);
+        $self->status_message('Found '. $individual_dna_sample_count .' samples for individual '. $individual->__display_name__);
+
+        if ( $individual_dna_sample_count > 2) {
+            $self->fatal_message('This command currently does not support more than 2 DNA samples per individual.');
+        }
+
+        my @normal_samples;
+        my @tumor_samples;
+
+        for my $individual_dna_sample (@individual_dna_samples) {
+            my $dna_sample_common_name = $individual_dna_sample->common_name || "NULL";
+            push(@normal_samples, $individual_dna_sample) if ($dna_sample_common_name =~ /$normal_sample_common_names/i);
+            push(@tumor_samples,  $individual_dna_sample) if ($dna_sample_common_name =~ /$tumor_sample_common_names/i);
```

Comment:
If I have two samples, one with common name `methodically collected normal` and one without a common name (thus `NULL`), this could lead to a subject mapping where we have one sample in both the tumor and normal position and the other sample silently omitted.